Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -658,6 +658,14 @@ protected class ClassWriter : IClassWriter {
       return TypeName(type, wr, tok, boxed, false, member);
     }
 
+    private string CharTypeName(bool boxed) {
+      if (UnicodeChars) {
+        return boxed ? "dafny.CodePoint" : "int";
```

Comment:
Why not `rune` or `int32`?

Reply:
Java doesn't have either, or any other such specialized type. :(